src/config_file.c

```diff
@@ -108,8 +108,12 @@ static int config_file_open(git_config_backend *cfg, git_config_level_t level, c
 	if ((res = git_config_entries_new(&b->entries)) < 0)
 		return res;
 
-	if (!git_path_exists(b->file.path))
+	if (!git_path_exists(b->file.path)) {
 		return 0;
+    }
+    if (p_access(b->file.path, R_OK) < 0) {
+        res = GIT_ENOTFOUND; // without this read check sandboxed applications on macOS can't open any repository, because `config_file_read()` below will return GIT_ERROR when it can't read the global /Users/username/.gitconfig file, and the upper layers will just completely abort on GIT_ERROR when loading a config file (which seems a bit extreme).
+    }
```

Comment:
Some style tips:

- we typically avoid braces around single-line statements
- we use tabs for indentation
- we only use C-style comments (`/* */`)

Simulating this behaviour with git is in fact quite trivial:

```bash
$ git config --list
user.name = Patrick Steinhardt
...
$ chmod u-r ~/.config/git/config
$ git config --list
```

So git continues to work just fine in case it can't read a given file and won't even raise a warning. So I'd argue that instead of setting `res = GIT_ENOTFOUND`, we should just `return 0;` directly similar to how we treat a non-existing path.

Reply:
Oh, and to avoid wasting your time, you should probably format the comment like following:

```C
if (p_access(b->file.path, R_OK) != 0) {
	/*
	 * Git will silently ignore any configuration files that are not readable,
	 * and so should we. This is also required for sandboxed applications on
	 * macOS, where the global git config file will not be readable by libgit2.
	 */
	return 0;
}
```